### PR TITLE
fix(subst): ignore broken symlinks

### LINKED ISSUE #9593: dune subst Error: stat(test): No such file or directory on semgrep
## Expected Behavior
should work as part of semgrep build in opam

## Actual Behavior

"Error: stat(test): No such file or directory"

```
dune subst  --debug
Error: stat(test): No such file or directory
Raised by primitive operation at Stdune__Path.stat_exn in file
  "otherlibs/stdune/src/path.ml", line 1322, characters 17-40
Called from Main__Subst.is_a_source_file in file "bin/subst.ml", line 19,
  characters 5-25
Called from Main__Subst.subst.(fun) in file "bin/subst.ml", line 360,
  characters 7-28
Called from Stdlib__List.iter in file "list.ml", line 110, characters 12-15               Called from Fiber__Core.O.(>>|).(fun) in file "vendor/fiber/src/core.ml",                   line 252, characters 36-41
Called from Fiber__Scheduler.exec in file "vendor/fiber/src/scheduler.ml",
```

## Reproduction

- PR with a reproducing test: 

1.
1.
1.

## Specifications

- Version of `dune` (output of `dune --version`):
- Version of `ocaml` (output of `ocamlc --version`)
- Operating system (distribution and version):

## Additional information 

- Link to gist with verbose output (run `dune` with the `--verbose` flag):

### PR DESCRIPTION
**Pull request by emillon**
Fixes #9593

### CODE REVIEW

File: bin/subst.ml [hunk 1]
```diff
@@ -16,7 +16,18 @@ let is_a_source_file path =
    | ".ttf"
    | ".woff" -> false
    | _ -> true)
-  && (Path.stat_exn path).st_kind = Unix.S_REG
+
+let is_kind_a_source_file path=
+  match Path.stat_exn path with
```
Comment (Leonidas-from-XIV) [inline bin/subst.ml:21]: Maybe use `Path.stat` instead of `stat_exn`?

**Review by Leonidas-from-XIV — APPROVED** [review]
Looks good and understandable, but needs to be reformatted.

File: bin/subst.ml (hunk 1 above)
Comment (emillon) [reply bin/subst.ml]: I'm a bit torn about this one this we still need to reraise the other error cases. I pushed a commit, let me know what you think.

Comment (Leonidas-from-XIV) [reply bin/subst.ml]: Ah, that's a good point. That said, I think the current state looks good to me and I am in favor of merging it.

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**